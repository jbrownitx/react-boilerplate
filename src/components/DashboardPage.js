import React from 'react';

export { DashboardPage as default };

const DashboardPage = () => (
	<div>
		Dashboard page.
	</div>
);