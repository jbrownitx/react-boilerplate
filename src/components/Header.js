import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export { Header, ConnectedHeader as default };

const Header = ({ startLogout, isPublic }) => (
	<header className="header">
		<div className="content-container">
			<Link className="header__title" to="/dashboard">
				<h1>Boilerplate</h1>
			</Link>&nbsp;|&nbsp;
			<button className="header__button" onClick={startLogout}>Log Out</button>
		</div>
	</header>
);

const mapDispatchToProps = dispatch => ({
	startLogout: () => dispatch(startLogout())
});

const ConnectedHeader = connect(undefined, mapDispatchToProps)(Header);