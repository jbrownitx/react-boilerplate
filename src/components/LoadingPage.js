import React from 'react';

export {LoadingPage as default};

const LoadingPage = () => (
	<div className="loader">
		<img className="loader__image" src="/images/loader.gif" />
	</div>
);