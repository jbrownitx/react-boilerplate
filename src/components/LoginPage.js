import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export { LoginPage, ConnectedLoginPage  as default };

const LoginPage = ({ startLogin }) => (
	<div className="box-layout">
		<div className="box-layout__box">
			<h1 className="box-layout__title">Boilerplate</h1>
			<p>Welcome message goes here.</p>
			<button className="box-layout__button" onClick={startLogin}>Log In with Google</button>
		</div>
	</div>
);

const mapDispatchToProps = dispatch => ({
	startLogin: () => dispatch(startLogin())
});

const ConnectedLoginPage = connect(undefined, mapDispatchToProps)(LoginPage);