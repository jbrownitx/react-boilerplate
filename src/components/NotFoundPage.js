import React from 'react';
import { Link } from 'react-router-dom';

export { NotFoundPage as default };

const NotFoundPage = () => (
	<div>
		<p>It's a 404 page!</p>
		<p>
			<Link to="/">Go home</Link>
		</p>
	</div>
);