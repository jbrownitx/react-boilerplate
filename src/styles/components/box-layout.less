.box-layout {
	background: url('/images/bg.jpg');
	background-size: cover;
	height: 100vh;
	width: 100vw;
	display: flex;
	align-items: center;
	justify-content: center;
}

.box-layout__box {
	border-radius: 3px;
	background-color: rgba(255, 255, 255, 0.85);
	text-align: center;
	width: 25rem;
	padding: @size-l @size-m;
}

.box-layout__title {
	margin: 0 0 @size-m 0;
	line-height: 1;
}

.box-layout__button {
	.button();
}