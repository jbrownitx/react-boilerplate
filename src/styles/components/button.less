.button() {
	background-color: @highlight-light;
	border: 0;
	color: @light-max;
	padding: @size-s;
	font-weight: 300;
	font-size: @size-font-l;
}

.button-link() {
	.button();

	display: inline-block;
	text-decoration: none;
	line-height: 1;
}

.button-alt {
	.button();

	background-color: #888;
}