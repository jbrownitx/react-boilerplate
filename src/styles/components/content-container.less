.content-container {
	max-width: 80rem;
	padding: 0 @size-m;
	margin: 0 auto;
}