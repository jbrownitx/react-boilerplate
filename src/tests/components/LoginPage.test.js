import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let startLogin, loginPage;

beforeEach(() => {
	startLogin = jest.fn();
	loginPage = shallow(<LoginPage startLogin={startLogin} />);
});

it('should render LoginPage correctly', () => {
	expect(loginPage).toMatchSnapshot();
});

it('should call startLogin when Log In button clicked', () => {
	loginPage.find('button').simulate('click');
	expect(startLogin).toBeCalled();
});