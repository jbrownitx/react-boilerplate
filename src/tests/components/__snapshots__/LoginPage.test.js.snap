// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      Welcome message goes here.
    </p>
    <button
      className="box-layout__button"
      onClick={[MockFunction]}
    >
      Log In with Google
    </button>
  </div>
</div>
`;
